stl/inc/xmemory

```diff
@@ -1593,10 +1642,22 @@ namespace ranges {
     uninitialized_move_result<_It, _Out> _Uninitialized_move_unchecked(
             _It _IFirst, _Se _ILast, _Out _OFirst, _OSe _OLast) {
         // clang-format on
+        constexpr bool _Is_sized1 = sized_sentinel_for<_Se, _It>;
+        constexpr bool _Is_sized2 = sized_sentinel_for<_OSe, _Out>;
         if constexpr (_Ptr_move_cat<_It, _Out>::_Really_trivial
-                      && sized_sentinel_for<_Se, _It> && sized_sentinel_for<_OSe, _Out>) {
-            return _Copy_memcpy_common(
-                _IFirst, _RANGES next(_IFirst, _STD move(_ILast)), _OFirst, _RANGES next(_OFirst, _STD move(_OLast)));
+                      && (_Is_sized1 || same_as<_Se, unreachable_sentinel_t>)
+#pragma warning(suppress : 6287) // Redundant code: the left and right subexpressions are identical
+                      &&(_Is_sized2 || same_as<_OSe, unreachable_sentinel_t>) ) {

```

Comment:
This is the third time this has popped up. I think it warrants to be extracted into a helper `_Is_sized_or_unreachable`